clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitJobExecutor.java

```diff
@@ -105,71 +107,99 @@ public void cloneOrPull(String repoUrl, String branch, Path localPath, String re
           repoUrl,
           branch,
           dotGitPath.toString());
-      clone(repoUrl, branch, localPath);
+      clone(repoUrl, branch, localPath, repoBasename);
       return;
     }
     // localPath has "<repo>/.git" directory
 
-    pull(repoUrl, dotGitPath.getParent());
+    pull(repoUrl, branch, dotGitPath.getParent());
   }
 
-  private void clone(String repoUrl, String branch, Path destination) throws IOException {
+  private void clone(String repoUrl, String branch, Path destination, String repoBasename)
+      throws IOException {
     if (!isValidReference(repoUrl)) {
       throw new IllegalArgumentException(
           "Git reference \""
               + repoUrl
               + "\" is invalid for credentials with auth type "
               + authType);
     }
-    FileUtils.deleteDirectory(destination.toFile());
+
+    if (destination.toFile().exists()) {
```

Comment:
Wasn't it fine before without the `if`? The first step of `FileUtils.deleteDirectory()` is to check if the directory exists. You could use a local variable for `destination.toFile()`.

Reply:
I only looked at the javadoc of that method and found this:

```
Throws IllegalArgumentException – if directory does not exist or is not a directory
```

So I just wanted to be sure. I'll use the local variable for `destination.toFile()`.